src/common/com/intellij/plugins/haxe/ide/annotator/HaxeAnnotatingVisitor.java

```diff
@@ -87,10 +91,74 @@ public void visitReferenceExpression(@NotNull HaxeReferenceExpression reference)
   @Override
+  public void visitFunctionDeclarationWithAttributes(@NotNull HaxeFunctionDeclarationWithAttributes functionDeclaration) {
+    List<HaxeCustomMeta> metas = functionDeclaration.getCustomMetaList();
+    for (HaxeCustomMeta meta : metas) {
+      if (HaxeModifierType.DEPRECATED.s.equals(meta.getText())) {
+        handleDeprecatedFunctionDeclaration(functionDeclaration);
+      }
+    }
+
+    super.visitFunctionDeclarationWithAttributes(functionDeclaration);
+  }
+
+  @Override
+  public void visitCallExpression(@NotNull HaxeCallExpression o) {
+    PsiElement child = o.getFirstChild();
+    if (child instanceof HaxeReferenceExpression) {
+      HaxeReferenceExpression referenceExpression = (HaxeReferenceExpression) child;
+      PsiElement reference = referenceExpression.resolve();
+
+      if (reference instanceof HaxeFunctionDeclarationWithAttributes) {
+        HaxeFunctionDeclarationWithAttributes functionDeclaration = (HaxeFunctionDeclarationWithAttributes)reference;
+
+        List<HaxeCustomMeta> metas = functionDeclaration.getCustomMetaList();
+        for (HaxeCustomMeta meta : metas) {
+          if (HaxeModifierType.DEPRECATED.s.equals(meta.getText())) {
+            handleDeprecatedCallExpression((HaxeReferenceExpression)referenceExpression.getLastChild());
+          }
+        }
+      }
+    }
+
+    super.visitCallExpression(o);
+  }
+
+  @Override
+  public void visitVarDeclaration(@NotNull HaxeVarDeclaration varDeclaration) {
+    List<HaxeCustomMeta> metas = varDeclaration.getCustomMetaList();
+    for (HaxeCustomMeta meta : metas) {
+      if (HaxeModifierType.DEPRECATED.s.equals(meta.getText())) {
+        handleDeprecatedVarDeclaration(varDeclaration);
+      }
+    }
+
+    super.visitVarDeclaration(varDeclaration);
+  }
+
+  @Override
   public void visitElement(PsiElement element) {
     ProgressIndicatorProvider.checkCanceled();
     element.acceptChildren(this);
   }
 
-  abstract protected void handleUnresolvedReference(HaxeReferenceExpression reference);
-}
+  protected void handleUnresolvedReference(HaxeReferenceExpression reference) {}
+  protected void handleDeprecatedFunctionDeclaration(HaxeFunctionDeclarationWithAttributes functionDeclaration) {}
+  protected void handleDeprecatedCallExpression(HaxeReferenceExpression referenceExpression) {}
+  protected void handleDeprecatedVarDeclaration(HaxeVarDeclaration varDeclaration) {}
+
+  private void checkDeprecatedVarCall(HaxeReferenceExpression referenceExpression) {
+    PsiElement reference = referenceExpression.resolve();
+
+    if (reference instanceof HaxeVarDeclarationPart) {
+      HaxeVarDeclaration varDeclaration = (HaxeVarDeclaration) reference.getParent();
+
+      List<HaxeCustomMeta> metas = varDeclaration.getCustomMetaList();
+      for (HaxeCustomMeta meta : metas) {
+        if (HaxeModifierType.DEPRECATED.s.equals(meta.getText())) {
+          handleDeprecatedCallExpression((HaxeReferenceExpression)referenceExpression.getLastChild());
```

Comment:
Looks like it works correctly.

[![https://gyazo.com/f7c250b775b4baac207160f3ffde8ce2](https://i.gyazo.com/f7c250b775b4baac207160f3ffde8ce2.png)](https://gyazo.com/f7c250b775b4baac207160f3ffde8ce2)